src/lib/operators/table_scan/column_is_null_table_scan_impl.cpp

```diff
@@ -112,42 +141,116 @@ void ColumnIsNullTableScanImpl::_scan_value_segment(const BaseValueSegment& segm
     return;
   }
 
-  DebugAssert(segment.is_nullable(), "Columns that are not nullable should have been caught by edge case handling.");
-
-  auto iterable = NullValueVectorIterable{segment.null_values()};
+  if constexpr (std::is_same_v<BaseSegmentType, hyrise::BaseValueSegment>) {
```

Comment:
Some comment/TODO that first and third branch could be merged it there was a uniform `null_values()` interface?